File: Samples/VB/08-Images/Form1.vb
Imports System.Text
Imports RibbonLib
Imports RibbonLib.Controls
Imports RibbonLib.Controls.Events
Imports RibbonLib.Interop

Namespace _08_Images
	Public Enum RibbonMarkupCommands As UInteger
		 cmdTabDrop = 1012
		 cmdGroupDrop = 1015
		 cmdButtonDropA = 1008
		 cmdButtonDropB = 1009
		 cmdButtonDropC = 1010
	End Enum

	Partial Public Class Form1
		Inherits Form
		Private exitOn As Boolean = False
		Private _tabDrop As RibbonTab
		Private _buttonDropA As RibbonButton
		Private _buttonDropB As RibbonButton
		Private _buttonDropC As RibbonButton

		Public Sub New()
			InitializeComponent()

			_tabDrop = New RibbonTab(_ribbon, CUInt(RibbonMarkupCommands.cmdTabDrop))
			_buttonDropA = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDropA))
			_buttonDropB = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDropB))
			_buttonDropC = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDropC))

			AddHandler _buttonDropA.ExecuteEvent, AddressOf _buttonDropA_ExecuteEvent
			AddHandler _buttonDropB.ExecuteEvent, AddressOf _buttonDropB_ExecuteEvent
		End Sub

		Private Sub _buttonDropA_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			' load bitmap from file
			Dim bitmap_Renamed As Bitmap = New Bitmap("..\..\Res\Drop32.bmp")
			bitmap_Renamed.MakeTransparent()

			' set large image property
			_buttonDropA.LargeImage = _ribbon.ConvertToUIImage(bitmap_Renamed)
		End Sub

		Private Sub _buttonDropB_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			Dim supportedImageSizes As New List(Of Integer)() From {32, 48, 64}

			Dim bitmap_Renamed As Bitmap
			Dim bitmapFileName As New StringBuilder()

			Dim selectedImageSize As Integer
			If supportedImageSizes.Contains(SystemInformation.IconSize.Width) Then
				selectedImageSize = SystemInformation.IconSize.Width
			Else
				selectedImageSize = 32
			End If

			exitOn = Not exitOn
			Dim exitStatus As String = If(exitOn, "on", "off")

			bitmapFileName.AppendFormat("..\..\Res\Exit{0}{1}.bmp", exitStatus, selectedImageSize)

			bitmap_Renamed = New Bitmap(bitmapFileName.ToString())
			bitmap_Renamed.MakeTransparent()

			_buttonDropB.LargeImage = _ribbon.ConvertToUIImage(bitmap_Renamed)
		End Sub
	End Class
End Namespace

File: Samples/CS/08-Images/Form1.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using RibbonLib;
using RibbonLib.Controls;
using RibbonLib.Controls.Events;
using RibbonLib.Interop;

namespace _08_Images
{
    public enum RibbonMarkupCommands : uint 
    {
         cmdTabDrop = 1012,
         cmdGroupDrop = 1015,
         cmdButtonDropA = 1008,
         cmdButtonDropB = 1009,
         cmdButtonDropC = 1010,
    }

    public partial class Form1 : Form
    {
        private bool exitOn = false;
        private RibbonTab _tabDrop;
        private RibbonButton _buttonDropA;
        private RibbonButton _buttonDropB;
        private RibbonButton _buttonDropC;

        public Form1()
        {
            InitializeComponent();

            _tabDrop = new RibbonTab(_ribbon, (uint)RibbonMarkupCommands.cmdTabDrop);
            _buttonDropA = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDropA);
            _buttonDropB = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDropB);
            _buttonDropC = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButtonDropC);

            _buttonDropA.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_buttonDropA_ExecuteEvent);
            _buttonDropB.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_buttonDropB_ExecuteEvent);
        }

        void _buttonDropA_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            // load bitmap from file
            Bitmap bitmap = new System.Drawing.Bitmap(@"..\..\Res\Drop32.bmp");
            bitmap.MakeTransparent();

            // set large image property
            _buttonDropA.LargeImage = _ribbon.ConvertToUIImage(bitmap);
        }

        void _buttonDropB_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            List<int> supportedImageSizes = new List<int>() { 32, 48, 64 };

            Bitmap bitmap;
            StringBuilder bitmapFileName = new StringBuilder();

            int selectedImageSize;
            if (supportedImageSizes.Contains(SystemInformation.IconSize.Width))
            {
                selectedImageSize = SystemInformation.IconSize.Width;
            }
            else
            {
                selectedImageSize = 32;
            }

            exitOn = !exitOn;
            string exitStatus = exitOn ? "on" : "off";

            bitmapFileName.AppendFormat(@"..\..\Res\Exit{0}{1}.bmp", exitStatus, selectedImageSize);

            bitmap = new System.Drawing.Bitmap(bitmapFileName.ToString());
            bitmap.MakeTransparent();

            _buttonDropB.LargeImage = _ribbon.ConvertToUIImage(bitmap);
        }
    }
}
